src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -16,10 +16,25 @@
 
 namespace Microsoft.Data.SqlClient
 {
+    internal readonly ref struct SessionHandle
+    {
+        public readonly SNIHandle NativeHandle;
+
+        public SessionHandle(SNIHandle nativeHandle) => NativeHandle = nativeHandle;
+        
+        public bool IsNull => NativeHandle is null;
+    }
+
     internal partial class TdsParserStateObject
     {
+        private static bool UseManagedSNI => false;
+
         private SNIHandle _sessionHandle = null;              // the SNI handle we're to work on
```

Comment:
Do you mean to replace SNIHandle with SessionHandle struct at line 32?

If yes, you can't do this because I have defined it as `readonly ref`. I needed to "add" the IsNull property to SNIHandle, so I encapsulated it in a struct. Then I added `readonly ref` so that this struct never allocates memory and is copied as if it was a plain reference. This is an optimization which is probably not needed, but I tried to make as little behavioral/performance changes as possible while refactoring.

Reply:
Ok I see. I was asking myself about encapsulating like in netcore version that why I have read it too fast. Thanks for clarification